### PR TITLE
Add Prow config for GoogleCloudPlatform/k8s-cloud-provider

### PR DESCRIPTION
**Pull request by rramkumar1**
/assign @BenTheElder

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/oss-test-infra/229)

### CODE REVIEW

File: prow/prowjobs/GoogleCloudPlatform/k8s-cloud-provider/OWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+bowei
```
[the remark this answers is not in the document]
Comment (BenTheElder) [reply prow/prowjobs/GoogleCloudPlatform/k8s-cloud-provider/OWNERS]:
```
approvers:
- bowei
- rramkumar1
...
```

[the remark this answers is not in the document]
Comment (rramkumar1) [reply prow/prowjobs/GoogleCloudPlatform/k8s-cloud-provider/OWNERS]: Done

**Comment by rramkumar1** [comment]
@bowei

**Comment by BenTheElder** [comment]
@cjwagner seems we're requiring people to use the pod-utils by way of the config validation enforcing warnings as errors?

**Comment by bowei** [comment]
This seems to fail the verify?

File: prow/prowjobs/GoogleCloudPlatform/k8s-cloud-provider/k8s-cloud-provider-config.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20191213-55437e3-master
+        args:
+        - "--repo=github.com/GoogleCloudPlatform/$(REPO_NAME)=$(PULL_REFS)"
+        - "--root=/go/src"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
+        - --scenario=execute
+        - --
+        - make
+        - --
+        - test
```
Comment (BenTheElder) [inline prow/prowjobs/GoogleCloudPlatform/k8s-cloud-provider/k8s-cloud-provider-config.yaml]: can switch to pod-utils with roughly
```yaml
presubmits:
  GoogleCloudPlatform/k8s-cloud-provider:
  - name: pull-k8s-cloud-provider-test
    branches:
    - master
    always_run: true         # Run for every PR, or only when requested.
    labels:
      preset-service-account: "true"
   decorate: true
    spec:
      containers:
      - image: gcr.io/k8s-testimages/kubekins-e2e:v20191213-55437e3-master
        command:
        - make
        - --
        - test
```

Comment (BenTheElder) [reply prow/prowjobs/GoogleCloudPlatform/k8s-cloud-provider/k8s-cloud-provider-config.yaml]: might want a `path_alias` to set the checkout to .../go/src/<alias>

Comment (BenTheElder) [reply prow/prowjobs/GoogleCloudPlatform/k8s-cloud-provider/k8s-cloud-provider-config.yaml]: https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md
https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md

Comment (BenTheElder) [reply prow/prowjobs/GoogleCloudPlatform/k8s-cloud-provider/k8s-cloud-provider-config.yaml]: sample https://github.com/kubernetes/test-infra/blob/b3462c342653e56c2b144ba1c7e7c98f169cb3c6/config/jobs/kubernetes/kops/kops-config.yaml#L15

**Pull request merged**